### PR TITLE
add lints for mao

### PR DESCRIPTION
**Pull request by enxebre**

### CODE REVIEW

**Comment by enxebre** [comment]
See https://github.com/openshift/machine-api-operator/pull/42
Ping @paulfantom @spangenberg @trawler

File: ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-presubmits.yaml [hunk 1]
```diff
@@ -62,6 +62,85 @@ presubmits:
...
+        - ./hack/yaml-lint.sh
+        command:
+        - sh
+        env:
+        - name: IS_CONTAINER
+          value: "TRUE"
+        image: quay.io/coreos/yamllint
+        name: ""
+        resources: {}
+    trigger: (?m)^/test yaml-lint
```
Comment (paulfantom) [inline ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-presubmits.yaml]: This way we won't retrigger linting on `/test all` comment, is it intentional?

File: ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-presubmits.yaml [hunk 2]
```diff
@@ -62,6 +62,85 @@ presubmits:
...
+        - .
+        command:
+        - sh
+        env:
+        - name: IS_CONTAINER
+          value: "TRUE"
+        image: quay.io/coreos/golang-testing
+        name: ""
+        resources: {}
+    trigger: (?m)^/test go-fmt
```
Comment (paulfantom) [inline ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-presubmits.yaml]: same as above

File: ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-presubmits.yaml [hunk 3]
```diff
@@ -62,6 +62,85 @@ presubmits:
...
+    always_run: true
+    context: ci/prow/go-vet
+    decorate: true
+    name: pull-ci-openshift-machine-api-operator-go-vet
+    rerun_command: /test go-vet
+    spec:
+      containers:
+      - args:
+        - ./hack/go-vet.sh
+        - ./...
```
Comment (paulfantom) [inline ci-operator/jobs/openshift/machine-api-operator/openshift-machine-api-operator-presubmits.yaml]: what's that?

**Pull request merged**